web/src/echo.ts

```diff
@@ -557,6 +557,7 @@ export function process_from_server(messages: ServerMessage[]): ServerMessage[]
                 // message content, but in practice, there's no harm to just
                 // doing it unconditionally.
                 msg_list.view.rerender_messages(msgs_to_rerender_or_add_to_narrow);
+                msg_list.add_messages(msgs_to_rerender_or_add_to_narrow);
```

Comment:
We also need to iterate over cached message list data objects here, I will add a commit for that in #32150 which has a bunch of related changes as well.

Reply:
OK, sounds good.